modules/saslserv/main.c

```diff
@@ -656,11 +622,11 @@ sasl_newuser(hook_user_nick_t *data)
 	p->flags &= ~ASASL_NEED_LOG;
 
 	/* Find the account */
-	myuser_t *const mu = p->authzid ? myuser_find_by_nick(p->authzid) : NULL;
+	myuser_t *const mu = p->authzeid ? myuser_find_uid(p->authzeid) : NULL;
 	if (! mu)
 	{
 		(void) notice(saslsvs->nick, u->nick, "Account %s dropped, login cancelled",
-		                                      p->authzid ? p->authzid : "??");
+		                                      p->authzeid ? p->authzeid : "??");
```

Comment:
As remarked on IRC, this should possibly indicate the client-specified authzid instead. We would probably need to keep track of it alongside the EID for logging purposes.